src/xml/builder.cr

```diff
@@ -360,7 +360,7 @@ module XML
   # Writes XML fragment into the given `IO`. An `XML::Builder` is yielded to the block.
   #
   # Builds an XML fragment without XML declaration (`<?xml?>`).
-  def self.build_fragment(io : IO, *, indent = nil, quote_char = nil)
+  def self.build_fragment(io : IO, *, indent = nil, quote_char = nil) : Nil
```

Comment:
If you are enforcing `nil` do you still need to return `v` at the end?

https://github.com/crystal-lang/crystal/pull/8876/files#diff-c34fb2ed7bd91eabe52d9954efba8a88R374

Reply:
Oh, I guess here we don't want `nil`, I'm not sure... I'll remove the `Nil` annotation

Reply:
I am surprised this compiled with the type restriction.

Reply:
The `Nil` type restriction forces the method to return `nil`, whatever it returns.

Reply:
I was a bit surprised about returning the yielded value when I visited this code in #8813. I suppose it's kind of nice, but I really doubt there are particular use cases for this. Returning `nil` would be fine, too. It's probably a question of continuity: should we expect all yielding methods that only have side effects and don't return anything inherently to return the block value? I guess this could get complicated when block execution is optional. Also, while it helps to simplify code that would otherwise rely on closured variables, I doubt there are many actually use cases where this comes in handy.

Reply:
Yeah, I think we should leave it as `Nil`